### ISSUE TITLE
Replace the simplify modules

### ISSUE THREAD

**Issue by 8hantanu** [aerospaceresearch/VisMa#142]
### To do:
Replace most of the methods (for add, sub, mul, div) in the **visma/simplify** module with class methods in function classes in **visma/functions**.

### Current implementation:
Go through the methods defined in **visma/simplify/***.
For example doing a simple operation like addition of two functions tokens would be something like:
```python
sumTok, _, _, _, _ = simplify([tok1, Binary('+'), tok2])
```
Here `tok1`, `tok2`, `sumTok` can be the subclasses of _visma.functions.structure.Function_ and, `Binary` and `simplify()` are imported from **visma.functions.operator** and **visma.simplify.simplify** respectively.

### Expected outcome:
The sum token must be simply updated by doing
```python
sumTok = tok1 + tok2
```
Replace the `expressionAddition/Subtraction()`(and other related methods) in **simplify/addsub.py** by adding `__add__` and `__sub__` class method to the various Function classes. Same goes for **simplify/muldiv.py**.

The following links might also be helpful:
- http://www.marinamele.com/2014/04/modifying-add-method-of-python-class.html
- https://www.python-course.eu/python3_magic_methods.php

**Note**: Please DONT add a wrapper function which can be something like this.
```python
def __add__(self, tok):
    sumTok, _, _, _, _ = simplify([self, Binary('+'), tok2])
    return sumTok
```
The main intention is to make the whole simplification modular and easy when implementing new function classes. 

### Extras:
- The methods in **matrix/operations.py** can be redefined as class methods of _Matrix_ class.
- Also expression operations can be redefined in _Expression_ class (#141)

**Comment by Alii-Khaled** [comment]
So here we will make operator overloading for ( + , - , * , / ) right ?!
what is the problem of the Wrapper functions i didn't understand the problem using them ?!
So if they are not the best solution, what should be the best code for addition for example ?!

**Comment by Alii-Khaled** [comment]
I have taken Object Oriented Programming course and i understand it well & i think i am up for the challenge of that Issue <3 .

**Comment by PulkitMishra** [comment]
@Alii-Khaled  are you working on this issue?
This looks like a mammoth task. I guess we should colab

**Comment by mayankDhiman** [comment]
I am working on this. @PulkitMishra. If you agree we can collab.

**Comment by Alii-Khaled** [comment]
Sure @PulkitMishra  we can collaborate <3 .

**Comment by Alii-Khaled** [comment]
we could talk in Visma's Zulip chat for it ?!

**Comment by abhishekti7** [comment]
@PulkitMishra @Alii-Khaled @mayankDhiman count me in too!

**Comment by Alii-Khaled** [comment]
Its will be cool we can Split work into 4 parts for now ( Addition , Subtraction , Multiplication , Division ) 👍
for me i have no problems to work on any one😄 .

**Comment by mayankDhiman** [comment]
Yep. Lets split it in 4 parts. @Alii-Khaled @zeus0789 @PulkitMishra. Zulip is ok.

**Comment by 8hantanu** [comment]
> So here we will make operator overloading for ( + , - , * , / ) right ?!

Yes.

> what is the problem of the Wrapper functions i didn't understand the problem using them ?!

The current implementation will require many changes in different methods(`getLevelVariables()`, `expressionAddtition` etc) and modules(simplify, io etc ) everytime a new Function type is added. So I thought rewriting the whole module will be much suitable for future development.

> So if they are not the best solution, what should be the best code for addition for example ?!

Its upto you to figure it out.

PRs for this can be made to the new branch([**simplify**](https://github.com/aerospaceresearch/visma/tree/simplify)).

**Comment by mayankDhiman** [comment]
Fixed with #197
